File: Problem 2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string input);
int count_words(string input);
int count_sentences(string input);

int main(void)
{
    //get string text
    string text = get_string("Text: ");
    //access the counters
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    //grade text
    int grade = round(0.0588 * (((float) letters / (float) words) * 100) - 0.296 * (((float) sentences / (float) words) * 100) - 15.8);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}

int count_letters(string input)
{
    int letters = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isalpha(input[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string input)
{
    int words = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isspace(input[i]))
        {
            words++;
        }
    }
    words ++;
    return words;
}

int count_sentences(string input)
{
    int sentences = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Problem 6/sentimental-readability/readability.py
from cs50 import get_string
# TODO
def main():
    text = get_string("Text: ")
    grade = index(text)
    if (grade < 1):
        print("Before Grade 1")
    elif (grade >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def letters(text):
    letter = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letter += 1
    return letter

def words(text):
    letter = text.split()
    word = len(letter)
    return word

def sentences(text):
    sentence = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            sentence += 1
    return sentence

def index(text):
    grade = round(0.0588 * (letters(text)/words(text))*100 - 0.296 * (sentences(text)/words(text))*100 - 15.8)
    return grade

main()